ERCS/erc-7390.md

```diff
@@ -0,0 +1,366 @@
+requires: 20
+---
+
+## Abstract
+
+This ERC defines a comprehensive set of functions and events facilitating seamless interactions (creation, management, exercising, etc.) for vanilla options. This standard ensures that options contracts conform to a common interface, facilitating the development of robust options trading platforms and enabling interoperability between dApps and protocols.
+
+## Motivation
+
+Options are widely used financial instruments that provide users with the right, but not the obligation, to buy or sell an underlying asset at a predetermined price within a specified timeframe. By introducing a standard interface for options contracts, we aim to foster a more inclusive and interoperable derivatives ecosystem on Ethereum. This standard will enhance the user experience and facilitate the development of decentralized options platforms, enabling users to seamlessly trade options across different applications.
+
+## Specification
+
+### Interface
+
+```solidity
+interface IERC7390 {
+    enum Side {
+        Call,
+        Put
+    }
+
+    struct VanillaOptionData {
+        Side side;
+        address underlyingToken;
+        uint256 amount;
+        address strikeToken;
+        uint256 strike;
+        address premiumToken;
+        uint256 premium;
+        uint256 exerciseWindowStart;
+        uint256 exerciseWindowEnd;
+    }
+
+    struct OptionIssuance {
+        VanillaOptionData data;
+        address seller;
+        uint256 exercisedOptions;
+        uint256 soldOptions;
+    }
+
+    event Created(uint256 indexed id);
+    event Bought(uint256 indexed id, uint256 amount, address indexed buyer);
+    event Exercised(uint256 indexed id, uint256 amount);
+    event Expired(uint256 indexed id);
+    event Canceled(uint256 indexed id);
+    event PremiumUpdated(uint256 indexed id, uint256 amount);
+
+    function create(VanillaOptionData calldata optionData) external returns (uint256);
+
+    function buy(uint256 id, uint256 amount) external;
+
+    function exercise(uint256 id, uint256 amount) external;
+
+    function retrieveExpiredTokens(uint256 id) external;
+
+    function cancel(uint256 id) external;
+
+    function updatePremium(uint256 id, uint256 amount) external;
+
+    function issuance(uint256 id) external view returns (OptionIssuance memory);
+}
+```
+
+### State Variable Descriptions
+
+At creation time, user must provide filled instance of `VanillaOptionData` structure that contains all the key information for initializing the option issuance.
+
+#### `side`
+
+**Type: `enum`**
+
+Side of the option. Can take the value `Call` or `Put`. `Call` option gives the option buyer right to exercise any acquired option tokens to buy the `underlying` token at given `strike` price using `strikeToken` from option seller. Similarly, `Put` option gives the option buyer right to sell the `underlying` token to the option seller at `strike` price.
+
+#### `underlyingToken`
+
+**Type: `address` (`IERC20`)**
```

Comment:
I'd expand on what you mean by "(`IERC20`)" somewhere. I think you're saying that `underlyingToken` should point to an ERC-20 contract, but you don't explicitly say that.